languages/go/TreeSitter/Go.hs

```diff
@@ -1,8 +1,15 @@
+{-# LANGUAGE TemplateHaskell #-}
 module TreeSitter.Go
 ( tree_sitter_go
+, Grammar(..)
 ) where
 
-import Foreign.Ptr
+import Language.Haskell.TH
+import TreeSitter.Go.Internal
 import TreeSitter.Language
 
-foreign import ccall unsafe "vendor/tree-sitter-go/src/parser.c tree_sitter_go" tree_sitter_go :: Ptr Language
+-- Regenerate template haskell code when these files change:
+addDependentFileRelative "../vendor/tree-sitter-go/src/parser.c"
+
+-- | Statically-known rules corresponding to symbols in the grammar.
+mkSymbolDatatype (mkName "Grammar") tree_sitter_go
```

Comment:
Each of the `tree-sitter-*` packages now defines a `Grammar` datatype for the language, per #111.